lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1125,15 +1125,18 @@ public function walkJoin($join)
                 $class      = $this->em->getClassMetadata($joinDeclaration->abstractSchemaName);
                 $dqlAlias   = $joinDeclaration->aliasIdentificationVariable;
                 $tableAlias = $this->getSQLTableAlias($class->table['name'], $dqlAlias);
-                $condition = '(' . $this->walkConditionalExpression($join->conditionalExpression) . ')';
+                $conditions = [];
+
+                if ($join->conditionalExpression) {
```

Comment:
@guilhermeblanco I was hunting a bug with that in the first place and then discovered that the parser does not implement `match()` correctly (which IMO would have made my original problem surface earlier).

I think I have provided some tests for that specific change, probably also described in #1328 and #1194.